spec.html

```diff
@@ -35917,15 +36218,33 @@ <h1>
         </emu-alg>
         <emu-grammar>Atom :: `.`</emu-grammar>
         <emu-alg>
-          1. Let _A_ be the CharSet of all characters.
+          1. Let _A_ be AllCharacters(_rer_).
           1. If _rer_.[[DotAll]] is not *true*, then
             1. Remove from _A_ all characters corresponding to a code point on the right-hand side of the |LineTerminator| production.
           1. Return CharacterSetMatcher(_rer_, _A_, *false*, _direction_).
         </emu-alg>
         <emu-grammar>Atom :: CharacterClass</emu-grammar>
         <emu-alg>
           1. Let _cc_ be CompileCharacterClass of |CharacterClass| with argument _rer_.
-          1. Return CharacterSetMatcher(_rer_, _cc_.[[CharSet]], _cc_.[[Invert]], _direction_).
+          1. Let _cs_ be _cc_.[[CharSet]].
+          1. If _rer_.[[UnicodeSets]] is *false*, or if every element of _cs_ consists of a single character, return CharacterSetMatcher(_rer_, _cs_, _cc_.[[Invert]], _direction_).
+          1. Assert: _cc_.[[Invert]] is *false*.
+          1. Let _lm_ be an empty List of Matchers.
+          1. For each string _s_ in _cs_ whose length is greater than 1, iterating in descending order of string length, do
+            1. Let _cs2_ be a one-element CharSet containing the last code point of _s_.
+            1. Let _m2_ be the result of CharacterSetMatcher(_rer_, _cs2_, *false*, _direction_).
+            1. For each code point _c1_ in _s_, iterating backwards from its second-to-last code point, do
+              1. Let _cs1_ be a one-element CharSet containing _c1_.
+              1. Let _m1_ be the result of CharacterSetMatcher(_rer_, _cs1_, *false*, _direction_).
+              1. Let _m2_ be MatchSequence(_m1_, _m2_, _direction_).
+            1. Append _m2_ to _lm_.
+          1. Let _singles_ be the CharSet containing every element of _cc_.[[CharSet]] which consists of a single character.
+          1. Append CharacterSetMatcher(_rer_, _singles_, *false*, _direction_) to _lm_.
+          1. If _cs_ contains the empty String, append EmptyMatcher() to _lm_.
```

Comment:
The capital 'S' on 'String' means that this is talking about an ES language value, which it probably shouldn't be.

```suggestion
          1. If _cs_ contains the empty sequence of characters, append EmptyMatcher() to _lm_.
```

Reply:
Thanks! Applied.